File: wallendar-app/app/src/main/java/com/td/wallendar/home/balances/BalanceViewHolder.java
package com.td.wallendar.home.balances;

import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.td.wallendar.R;
import com.td.wallendar.models.Debt;

import java.text.DecimalFormat;

public class BalanceViewHolder extends RecyclerView.ViewHolder {
    private static final DecimalFormat df = new DecimalFormat("0.00");

    private OnBalanceSettleUpClickedListener onBalanceSettleUpClickedListener;
    private OnRemindButtonClickedListener onRemindButtonClickedListener;

    public BalanceViewHolder(@NonNull View itemView) {
        super(itemView);
    }

    public void bind(final Debt debt, final long loggedUserId) {
        final TextView textView = itemView.findViewById(R.id.row_balance_text);
        final Button remindButton = itemView.findViewById(R.id.remind_debt_button);
        final Button settleUpButton = itemView.findViewById(R.id.settle_up_debt_button);

        String amount = df.format(debt.getAmount());
        if (debt.getFrom().getId() == loggedUserId) {
            remindButton.setVisibility(View.GONE);
            String userToText = debt.getTo().getFirstName() + " " + debt.getTo().getLastName();
            textView.setText(itemView.getContext().getString(R.string.you_owe_to_user_amount, userToText, amount));
        } else {
            String userFromText = debt.getFrom().getFirstName() + " " + debt.getFrom().getLastName();
            textView.setText(itemView.getContext().getString(R.string.user_owes_you_amount, userFromText, amount));
        }

        remindButton.setOnClickListener(view ->
                onRemindButtonClickedListener.onRemindButtonClick(debt));

        settleUpButton.setOnClickListener(view -> onBalanceSettleUpClickedListener.onBalanceSettleUpClick(debt));
    }

    public void setOnBalanceSettleUpClickedListener(OnBalanceSettleUpClickedListener listener) {
        this.onBalanceSettleUpClickedListener = listener;
    }

    public void setOnRemindButtonClickedListener(OnRemindButtonClickedListener listener) {
        this.onRemindButtonClickedListener = listener;
    }

}

File: wallendar-app/app/src/main/java/com/td/wallendar/home/balances/BalanceViewHolder.kt
package com.td.wallendar.home.balances

import android.view.View
import android.view.View.GONE
import android.view.View.VISIBLE
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.td.wallendar.R
import com.td.wallendar.models.Debt
import java.text.DecimalFormat

class BalanceViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    private var onBalanceSettleUpClickedListener: OnBalanceSettleUpClickedListener? = null
    private var onRemindButtonClickedListener: OnRemindButtonClickedListener? = null
    private var onDetailDebtClickedListener: OnDetailDebtClickedListener? = null
    fun bind(debt: Debt, loggedUserId: Long) {
        val textView = itemView.findViewById<TextView?>(R.id.row_balance_text)
        val remindButton = itemView.findViewById<Button?>(R.id.remind_debt_button)
        val settleUpButton = itemView.findViewById<Button?>(R.id.settle_up_debt_button)
        val debtDetail = itemView.findViewById<ImageView?>(R.id.debt_detail)

        if (!debt.containsDetails) {
            debtDetail.visibility = GONE
        } else {
            debtDetail.visibility = VISIBLE
            debtDetail.setOnClickListener { onDetailDebtClickedListener?.onDetailDebtClickedListener(debt.id) }
        }

        val amount = df.format(debt.amount)
        if (debt.from.id == loggedUserId) {
            remindButton.visibility = GONE
            val userToText = debt.to.firstName + " " + debt.to.lastName
            textView.text = itemView.context.getString(R.string.you_owe_to_user_amount, userToText, amount)
        } else {
            val userFromText = debt.from.firstName + " " + debt.from.lastName
            textView.text = itemView.context.getString(R.string.user_owes_you_amount, userFromText, amount)
        }
        remindButton.setOnClickListener { view: View? -> onRemindButtonClickedListener?.onRemindButtonClick(debt) }
        settleUpButton.setOnClickListener { view: View? -> onBalanceSettleUpClickedListener?.onBalanceSettleUpClick(debt) }
    }

    fun setOnBalanceSettleUpClickedListener(listener: OnBalanceSettleUpClickedListener?) {
        onBalanceSettleUpClickedListener = listener
    }

    fun setOnRemindButtonClickedListener(listener: OnRemindButtonClickedListener?) {
        onRemindButtonClickedListener = listener
    }

    fun setOnDetailDebtClickedListener(listener: OnDetailDebtClickedListener?) {
        onDetailDebtClickedListener = listener
    }

    companion object {
        private val df: DecimalFormat = DecimalFormat("0.00")
    }
}